types.go

```diff
@@ -387,23 +387,18 @@ type TableWritable interface {
 //
 
 const (
-	PrimaryIndexPos = 0
-
-	reservedIndexPrefix       = "__"
-	RevisionIndex             = "__revision__"
-	RevisionIndexPos          = 1
-	GraveyardIndex            = "__graveyard__"
-	GraveyardIndexPos         = 2
-	GraveyardRevisionIndex    = "__graveyard_revision__"
-	GraveyardRevisionIndexPos = 3
-
-	SecondaryIndexStartPos = 4
+	PrimaryIndexPos        = 0
+	reservedIndexPrefix    = "__"
+	RevisionIndex          = "__revision__"
+	RevisionIndexPos       = 1
+	SecondaryIndexStartPos = 2
 )
 
 // object is the format in which data is stored in the tables.
 type object struct {
 	revision uint64
 	data     any
+	deleted  bool
```

Comment:
I played a bit with having `data` point to `type struct deleted { data any }`, but that just caused a further allocation so was no good. If we mandated that `T` is always a pointer then we could store `data unsafe.Pointer` here and use the bytes saved from the not having the type information around, but that's probably a bit annoying to work with. 

But yeah, probably worth checking the field alignments, especially with the object embedded into a part node. Will take a look.

Reply:
could also halve the revision space and use the bit 😇 
but I think you mentioned somewhere that the size didn't increase? that's surprising 😅

Reply:
Hmm, we could do that, but need to make sure the "deleted bit" is the least significant one which is a bit annoying (shift left by one and set). But yeah since the size didn't increase (most likely because there were padding in the node structs) not worth optimizing.